core/src/test/java/org/apache/iceberg/PartitionStatsHandlerTestBase.java

```diff
@@ -441,6 +441,52 @@ public void testPartitionStats() throws Exception {
             snapshot1.snapshotId()));
   }
 
+  @Test
+  public void testCopyOnWriteDelete() throws Exception {
+    Table testTable =
+        TestTables.create(tempDir("my_test"), "my_test", SCHEMA, SPEC, 2, fileFormatProperty);
+
+    DataFile dataFile1 =
+        DataFiles.builder(SPEC)
+            .withPath("/df1.parquet")
+            .withPartitionPath("c2=a/c3=a")
+            .withFileSizeInBytes(10)
+            .withRecordCount(1)
+            .build();
+    DataFile dataFile2 =
+        DataFiles.builder(SPEC)
+            .withPath("/df2.parquet")
+            .withPartitionPath("c2=b/c3=b")
+            .withFileSizeInBytes(10)
+            .withRecordCount(1)
+            .build();
+
+    testTable.newAppend().appendFile(dataFile1).appendFile(dataFile2).commit();
+
+    PartitionStatisticsFile statisticsFile =
+        PartitionStatsHandler.computeAndWriteStatsFile(testTable);
+    testTable.updatePartitionStatistics().setPartitionStatistics(statisticsFile).commit();
+
+    assertThat(
+            PartitionStatsHandler.readPartitionStatsFile(
+                PartitionStatsHandler.schema(Partitioning.partitionType(testTable)),
+                Files.localInput(statisticsFile.path())))
+        .allMatch(s -> (s.dataRecordCount() != 0 && s.dataFileCount() != 0));
+
+    testTable.newDelete().deleteFile(dataFile1).commit();
+    testTable.newDelete().deleteFile(dataFile2).commit();
+
+    PartitionStatisticsFile statisticsFileNew =
+        PartitionStatsHandler.computeAndWriteStatsFile(testTable);
+
+    // stats must be decremented to zero as all the files removed from table.
+    assertThat(
+            PartitionStatsHandler.readPartitionStatsFile(
+                PartitionStatsHandler.schema(Partitioning.partitionType(testTable)),
+                Files.localInput(statisticsFileNew.path())))
```

Comment:
BTW, I noted Files.localInput is also used in production code [PartitionStatsHandler::computeAndMergeStatsIncremental](https://github.com/apache/iceberg/blob/cab0decbb0e32bf314039e30807eb033c50665d5/core/src/main/java/org/apache/iceberg/PartitionStatsHandler.java#L279) to load the previous stats file. Shouldn't we use table's FileIO to do that?

Reply:
done